### DISCUSSION TITLE
Xournal++ works on android

### DISCUSSION THREAD

**Discussion by ipesando** [xournalpp/xournalpp#5654] [Show and tell] [answered]
Hi, 
thanks to termux and temux-x11 (https://github.com/termux/termux-x11) it is possible to have a usable xournal++ on android.
It is not perfect but usable.
The main issue is the appearance of random lines.
This can be partially be tackled by choosing "Thick" as pen default and "Mouse" as input, at least for my Samsung Galaxy Tab S6.
There are still some random lines but my experience is the following.
If they disappear with an undo, you can "reset" by selecting Eraser and then the Pen again.
Sometimes even when the Eraser  is selected signs and  lines appear when moving the pointer, in this case one may try by using the selection tool to select and ct the affected area.

Hope it help.
Thank for the work on xournal++

**Comment by rolandlo** [comment, marked as answer]
Oh wow, how's that even possible?! Can you give instructions how to run Xournal++ on Android, so other users can try it out as well? Maybe we find a way to fix the random lines issue.

**Comment by ipesando** [comment]
More details (I hope I haven't forgot anything)
1) Install termux from Play Store 
2) Install the app temux-x11 by downloading and installing the apk 
from https://github.com/termux/termux-x11/releases
(see also https://github.com/termux/termux-x11 for more options)
3) open termux and get the linux shell then 
pkg install x11-repo
pkg install xfce4
pkg install xournalpp
pkg install termux-x11-nightly
(it requires some space...)
4) start the app termux-x11
5) switch to termux again and from the shell
termux-x11 :1 -xstartup "dbus-launch --exit-with-session xfce4-session"
6) switch to termux-x11 and you see the xfce4 window manager
a) start the x-terminal
b) use the android "back" to get the keyboard
c) launch xournalpp from the shell of the x-terminal
d) customize xournalpp: default size->thick, input->mouse

I tried also with changing the pressure sensitivity for avoiding artifacts but without luck.

**Reply by rolandlo** [reply]
I got it to install on my Android phone (Galaxy Samsung S8) as well. One more step (before step 5) was needed for me before termux-x11 got connection:
```
apt update && apt upgrade
```

I'm amazed that it runs. However running it on a phone (which is pretty small) is not a great user experience. I don't even have a stylus to draw. On an Android tablet it may be much more enyojable.

**Reply by Mik-el** [reply]
It works for me, but I can't start audio recordings, do I have to install something else?

**Reply by rolandlo** [reply]
> It works for me, but I can't start audio recordings, do I have to install something else?

I don't know. Have you played with the input and output device settings in menu `Edit > Preferences`? What are your options there?

**Reply by m1neral-rocks** [reply]
Oh wow I just tested this on a medium-sized samsung pen tablet and it works really well. I'm not seeing any artifacts, there's no latency and the pen pressure works perfectly fine. It just made me want a decent android port of xournalpp more so I could finally ditch samsung notes.
